### PR TITLE
fix: [CDS-36473]: fixed time filter reset issue for (home, services, overview,serviceDetail) dashboards

### PR DESCRIPTION
**Pull request by Sagar-Rajput7**
##### Summary:

> When the user navigates between different views (Services, Overview, Home,serviceDetail) the system doesn’t remember the time filter that was selected (within the same session), and every time the user goes back to the same view, the filter resets.

> stored key-value pair for timeRange on sessionStorage (using existing useLocalStorageHook )
> sessionStorage will clear on re-login/new-login, tab close, origin (https, port etc. ) change etc.

##### Screenshots:

https://user-images.githubusercontent.com/98325173/169822533-1ee766c7-8dc8-4d2d-b7d8-e455a844fa43.mov

https://user-images.githubusercontent.com/98325173/169822813-bde0f8aa-f344-462e-9dbf-ccef57a3cadd.mov

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by Sagar-Rajput7** [comment]
retrigger jest

File: src/modules/75-cd/components/ServiceDetails/ServiceDetailsContent/ServiceDetailsSummary.tsx [hunk 1]
```diff
@@ -22,23 +22,35 @@ import { DeploymentsWidget } from '@cd/components/Services/DeploymentsWidget/Dep
...
+  )
+
+  const timeServiceDetail = validTimeFormat(timeRangeService)
+  timeRangeService.range[0] = timeServiceDetail.range[0]
+  timeRangeService.range[1] = timeServiceDetail.range[1]
   return (
     <Page.Body>
       <Container flex className={css.timeRangeContainer}>
-        <TimeRangeSelector timeRange={timeRange?.range} setTimeRange={setTimeRange} />
+        <TimeRangeSelector timeRange={timeRangeService?.range} setTimeRange={setTimeRangeService} />
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/ServiceDetails/ServiceDetailsContent/ServiceDetailsSummary.tsx:48]: You can choose to ignore this. But, I think serviceTimeRange conveys a better meaning or timeRangeService. When we look for a property of range, we would look at **TimeRange** instead of **Service**. However, timeRangeForService would convey it properly.

**Review by AshwinBhatkal-Harness — APPROVED** [review]

File: src/modules/75-cd/components/ServiceDetails/ServiceDetailsContent/ServiceDetailsSummary.tsx (hunk 1 above)
Comment (Sagar-Rajput7) [reply src/modules/75-cd/components/ServiceDetails/ServiceDetailsContent/ServiceDetailsSummary.tsx:48]: ok i will change in other PR which I will raise today itself - as that pr will also be depending on this.
Thanks for the suggestion.

**Review by satyam-harness — APPROVED** [review]

File: src/framework/utils/SecureStorage.ts [hunk 2]
```diff
@@ -33,6 +33,7 @@ export default class SecureStorage {
     }
 
     localStorage.clear()
+    sessionStorage.clear()
```
Comment (kapilkumawat86) [inline src/framework/utils/SecureStorage.ts:36]: why this change, please add coment

**Review by kapilkumawat86 — APPROVED** [review]

File: src/framework/utils/SecureStorage.ts (hunk 2 above)
Comment (Sagar-Rajput7) [reply src/framework/utils/SecureStorage.ts:36]: This is for clearing the sessionStorage on logout ( this is required because we want our time filters to reset to default when a new login session is triggered and for the rest, it should follow the default sessionStorage behavior )
(as existing sessionStorage behavior does not care about the login/logout of our site - but we want to do this from our end)

I will add the comment in the code -> in the next pr that I will be raising after this as that will be dependent on this.
will that be ok?
